Core.ApplicationServices/HelpTexts/HelpTextApplicationService.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Core.Abstractions.Types;
+using Core.ApplicationServices.Authorization;
+using Core.ApplicationServices.Extensions;
+using Core.ApplicationServices.Model.HelpTexts;
+using Core.DomainModel;
+using Core.DomainModel.Events;
+using Core.DomainServices;
+
+namespace Core.ApplicationServices.HelpTexts
+{
+    public class HelpTextApplicationService: IHelpTextApplicationService
+    {
+        private readonly IGenericRepository<HelpText> _helpTextsRepository;
+        private readonly IOrganizationalUserContext _userContext;
+        private readonly IDomainEvents _domainEvents;
+        private readonly IHelpTextService _helpTextService;
+
+        public HelpTextApplicationService(IGenericRepository<HelpText> helpTextsRepository,
+            IOrganizationalUserContext userContext, IDomainEvents domainEvents, IHelpTextService helpTextService)
+        {
+            _helpTextsRepository = helpTextsRepository;
+            _userContext = userContext;
+            _domainEvents = domainEvents;
+            _helpTextService = helpTextService;
+        }
+
+        public Result<HelpText, OperationError> GetHelpText(string key)
+        {
+            var helpText = this._helpTextsRepository.AsQueryable().FirstOrDefault(ht => ht.Key == key);
+            return helpText != null
+                ? helpText
+                : new OperationError($"Could not find help text with key {key}", OperationFailure.NotFound);
+        }
+
+        public Result<IEnumerable<HelpText>, OperationError> GetHelpTexts()
+        {
+            return Result<IEnumerable<HelpText>, OperationError>.Success(this._helpTextsRepository.AsQueryable().ToList());
+        }
+
+        public Result<HelpText, OperationError> CreateHelpText(HelpTextCreateParameters parameters)
+        {
+            return WithGlobalAdminRights("User is not allowed to create help text")
+                .Match(error => error,
```

Comment:
here again I would extract the Match to outside of the Match:
like this
![image](https://github.com/user-attachments/assets/f7c424b1-c10e-4d66-ba79-a9e85860539a)